unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/Passport/Passport/PassportNavigation/PassportNavigationComponentController.cs

```diff
@@ -11,32 +13,55 @@ namespace DCL.Social.Passports
     public class PassportNavigationComponentController
     {
         private const int MAX_NFT_COUNT = 40;
+
         private readonly IProfanityFilter profanityFilter;
         private readonly IWearableItemResolver wearableItemResolver;
         private readonly IWearableCatalogBridge wearableCatalogBridge;
         private readonly IEmotesCatalogService emotesCatalogService;
+        private readonly INamesService namesService;
+        private readonly ILandsService landsService;
+        private readonly IUserProfileBridge userProfileBridge;
         private readonly DataStore dataStore;
+        private string currentUserId;
 
+        private UserProfile ownUserProfile => userProfileBridge.GetOwn();
         private readonly IPassportNavigationComponentView view;
         private HashSet<string> cachedAvatarEquippedWearables = new ();
         private readonly List<string> loadedWearables = new List<string>();
-        public event Action<string> OnClickBuyNft;
+        public event Action<string, string> OnClickBuyNft;
+        public event Action OnClickCollectibles;
 
-        public PassportNavigationComponentController(IPassportNavigationComponentView view, IProfanityFilter profanityFilter, IWearableItemResolver wearableItemResolver, IWearableCatalogBridge wearableCatalogBridge, IEmotesCatalogService emotesCatalogService, DataStore dataStore)
+        public PassportNavigationComponentController(
+            IPassportNavigationComponentView view,
+            IProfanityFilter profanityFilter,
+            IWearableItemResolver wearableItemResolver,
+            IWearableCatalogBridge wearableCatalogBridge,
+            IEmotesCatalogService emotesCatalogService,
+            INamesService namesService,
+            ILandsService landsService,
+            IUserProfileBridge userProfileBridge,
+            DataStore dataStore)
         {
             this.view = view;
             this.profanityFilter = profanityFilter;
             this.wearableItemResolver = wearableItemResolver;
             this.wearableCatalogBridge = wearableCatalogBridge;
             this.emotesCatalogService = emotesCatalogService;
+            this.namesService = namesService;
+            this.landsService = landsService;
+            this.userProfileBridge = userProfileBridge;
             this.dataStore = dataStore;
-            view.OnClickBuyNft += (wearableId) => OnClickBuyNft?.Invoke(wearableId);
+            view.OnClickBuyNft += (wearableId, wearableType) => OnClickBuyNft?.Invoke(wearableType is "name" or "parcel" or "estate" ? currentUserId : wearableId, wearableType);
+            view.OnClickCollectibles += () => OnClickCollectibles?.Invoke();
         }
 
         public void UpdateWithUserProfile(UserProfile userProfile) => UpdateWithUserProfileAsync(userProfile).Forget();
```

Comment:
No reason to launch a separate flow as you call this method from another async flow. Make it return `UniTask` and await from the caller accordingly